docs/core_docs/docs/tutorials/chatbot.ipynb

```diff
@@ -526,7 +526,8 @@
     "import type { BaseMessage } from \"@langchain/core/messages\";\n",
     "import { RunnablePassthrough, RunnableSequence } from \"@langchain/core/runnables\";\n",
     "\n",
-    "const filterMessages = ({ chat_history }: { chat_history: BaseMessage[]; }) => {\n",
+    "const filterMessages = (input: Record<string, unknown>) => {\n",
+    "  const { chat_history } = input as { chat_history: BaseMessage[] };\n",
```

Comment:
```suggestion
    "  const { chat_history }: { chat_history: BaseMessage[] } = input;\n",
```

Reply:
@bracesproul The previous suggestion (from unknown to any) will work fine. But, this line will cause the same issue. 

```
Property 'chat_history' is missing in type 'Record<string, any>' but required in type '{ chat_history: BaseMessage[]; }'.ts(2741)
tutorial2_updated.ts(31, 31): 'chat_history' is declared here.
```

So, I believe we should have it as:

```js
const { chat_history } = input as { chat_history: BaseMessage[] };
```

Reply:
Jeez you're right, that's annoying. I pushed a commit doing this, which is a little more concise and does not throw any type errors `const filterMessages = (input: { chat_history: BaseMessage[] }) => input.chat_history.slice(-10);`

Reply:
@bracesproul  Actually, even with the new line, the code will still cause build error. But, in a different place. If you add the new line, it will error out in the following lines:

```js
const chain = RunnableSequence.from([
    RunnablePassthrough.assign({
      chat_history: filterMessages,  // <-- Error here
    }),
    prompt,
    model,
  ]).pipe(parser);
```


The following error will be reported:

```
Type '(input: { chat_history: BaseMessage[]; }) => BaseMessage[]' is not assignable to type 'RunnableLike<Record<string, unknown>, BaseMessage[]>'.
  Type '(input: { chat_history: BaseMessage[]; }) => BaseMessage[]' is not assignable to type 'RunnableFunc<Record<string, unknown>, BaseMessage[]>'.
    Types of parameters 'input' and 'input' are incompatible.
      Property 'chat_history' is missing in type 'Record<string, unknown>' but required in type '{ chat_history: BaseMessage[]; }'.ts(2322)
tutorial2_updated.ts(30, 36): 'chat_history' is declared here.
tutorial2_updated.ts(34, 7): The expected type comes from property 'chat_history' which is declared here on type 'RunnableMapLike<Record<string, unknown>, { chat_history: BaseMessage[]; }>'
```

Reply:
Thanks for notifying me of this. I'll just push a larger commit adding proper types throughout this entire doc. Give me a few min to do so.